src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs

```diff
@@ -11,7 +11,134 @@ namespace System.Net
 {
     public partial class WebProxy : IWebProxy, ISerializable
     {
-        private ArrayList? _bypassList;
+
+        private sealed class DirtyTrackingArrayList : ArrayList
```

Comment:
Also, let's call it ChangeTrackingArrayList instead of DirtyTrackingArrayList, use IsChanged instead of IsDirty, etc.